scalameter-core/src/main/scala/org/scalameter/utils/ClassPath.scala

```diff
@@ -8,51 +8,73 @@ import org.apache.commons.lang3.SystemUtils
+    case _ => false
+  }
+
+  override def hashCode(): Int = paths.hashCode()
+}
+
+object ClassPath {
+  private def validate(elem: String) = {
+    require(!elem.contains(File.pathSeparatorChar), s"Classpath library contains illegal character: ${File.pathSeparatorChar}")
+    if (SystemUtils.IS_OS_WINDOWS) require(!elem.contains("\""), "Classpath library contains illegal character: \"")
   }
 
-  /** Returns the default classpath string, surrounded by quotes on Windows.
+  private def fromString(classPath: String): ClassPath = {
+    val cl = if (SystemUtils.IS_OS_WINDOWS) classPath.stripPrefix("\"").stripSuffix("\"") else classPath
+    ClassPath(cl.split(File.pathSeparator).toList)
+  }
+
+  /** Constructs [[ClassPath]] from given list of strings validating them first.
    */
-  def platformSpecificDefault: String = {
-    platformSpecificExtract(this.getClass.getClassLoader, sys.props("java.class.path"))
+  def apply(paths: List[String]): ClassPath = {
+    paths.foreach(validate)
+    new ClassPath(paths)
   }
 
-  /** Same as `extract`, but returns a string surrounded with quotes on Windows.
+  /** Returns the default classpath string.
    */
-  def platformSpecificExtract(classLoader: ClassLoader, default: =>String): String = {
-    val cp = extract(classLoader, default)
-    if (SystemUtils.IS_OS_WINDOWS) "\"" + cp + "\"" else cp
+  def default: ClassPath = {
+    extract(this.getClass.getClassLoader, sys.props("java.class.path"))
   }
 
   /** Extracts the classpath from the given `classLoader` if it is a `URLClassLoader` or
    *  from the first parent that is a `URLClassLoader`.
    *  If no `URLClassLoader` can be found, returns the `default` classpath.
    */
-  def extract(classLoader: ClassLoader, default: =>String): String =
+  def extract(classLoader: ClassLoader, default: =>String): ClassPath =
     classLoader match {
-      case urlclassloader: URLClassLoader => extractFromUrlCL(urlclassloader)
-      case null => sys.props("sun.boot.class.path")
+      case urlclassloader: URLClassLoader => ClassPath(extractFileClasspaths(urlclassloader.getURLs))
+      case null => fromString(sys.props("sun.boot.class.path"))
       case _ =>
         val parent = classLoader.getParent
         if (parent != null)
           extract(parent, default)
         else
-          default
+          fromString(default)
     }
 
-  private def extractFromUrlCL(urlclassloader: URLClassLoader): String = {
-    val files = extractFileClasspaths(urlclassloader.getURLs)
-    files.mkString(File.pathSeparator)
-  }
-
-  private[scalameter] def extractFileClasspaths(urls: Seq[URL]): Seq[String] = {
+  private[scalameter] def extractFileClasspaths(urls: Seq[URL]): List[String] = {
     val fileResource = "file:(.*)".r
-    urls.map(s => URLDecoder.decode(s.toString, "UTF-8")) collect {
+    (urls.map(s => URLDecoder.decode(s.toString, "UTF-8")) collect {
       case orig @ fileResource(file) => file
-    }
+    })(collection.breakOut)
```

Comment:
I've just used `List[String]` in order to avoid introducing another pickler for `Seq[String]`. 
I know, that is simply `object SeqListPickler extends TraversablePickler[Seq, String]` but I thought that I should avoid it ;-)
I will change it.